### ISSUE TITLE
Compiling v4.19-rc1 fails due to missing __naked preprocessor fine

### ISSUE THREAD

**Issue by agners** [ClangBuiltLinux/linux#34]
Compiling Linux v4.19-rc1 for ARM32 fails due to missing __naked preprocessor define.

```
 $ make CC=clang HOSTCC=clang
  CALL    scripts/checksyscalls.sh
<stdin>:1332:2: warning: syscall io_pgetevents not implemented [-W#warnings]
#warning syscall io_pgetevents not implemented
 ^
1 warning generated.
  CHK     include/generated/compile.h
  CC      arch/arm/mach-exynos/mcpm-exynos.o
arch/arm/mach-exynos/mcpm-exynos.c:193:13: error: variable has incomplete type 'void'
static void __naked exynos_pm_power_up_setup(unsigned int affinity_level)
            ^
```

**Comment by agners** [comment]
Miguel Ojeda's Compiler Attributes patchset addresses this. However, ideally the relevant patches (5 and 6) should go into v4.19...

https://lore.kernel.org/lkml/20180908212459.19736-1-miguel.ojeda.sandonis@gmail.com/

**Comment by nickdesaulniers** [comment]
Hey @ojeda, land your patches! :full_moon_with_face:

**Comment by ojeda** [comment]
Done :)

**Comment by nickdesaulniers** [comment]
Link?

**Comment by nickdesaulniers** [comment]
@ojeda has posted his 4.19 series to lkml: https://lore.kernel.org/r/20180918165542.4691-1-miguel.ojeda.sandonis@gmail.com/

**Comment by ojeda** [comment]
> Link?

Oops, forgot to reply to this, sorry.

> @ojeda has posted his 4.19 series to lkml: https://lore.kernel.org/r/20180918165542.4691-1-miguel.ojeda.sandonis@gmail.com/

Yeah, that is the v2 (same as v1 plus the acks).

**Comment by dileks** [comment]
Through which submaintainer tree will this go through? linux-kbuild Git?

**Comment by ojeda** [comment]
> Through which submaintainer tree will this go through? linux-kbuild Git?

Greg will probably pick it up directly.

**Comment by dileks** [comment]
Compiler Attributes: naked was fixed in gcc 4.6
https://git.kernel.org/linus/d124b44f09cab67fc6da4a4513417e3e54b01efc

Compiler Attributes: naked can be shared
https://git.kernel.org/linus/ae596de1a0c8c2c924dc99d23c026259372ab234

**Comment by nickdesaulniers** [comment]
Thanks @dileks for the heads up + shas! Thanks @ojeda for the patches.  This should be fixed in the next tag after v4.19-rc4.
